### DISCUSSION TITLE
Is it possible to do a multiple BLE advertising on ESP32?

### DISCUSSION THREAD

**Discussion by amplatzer** [micropython/micropython#12881] [ESP32] [answered]
Basically, I want to advertise 2 or more GAP_NAME, and each of them has their own services.  Can I do that with ESP32 or ESP32S3?

**Comment by puppet13th** [comment]
i just tried it using aioble module, only the last one( named ESP32-2 ) was shown when scanning for peripheral.

```
asyncio.create_task( peripheral( name='ESP32-1' ) )
asyncio.run( peripheral( name='ESP32-2 ) )
```

**Comment by jimmo** [comment]
The gap name is actually distinct from the name in the advertising payload. (The former is what you return from the name characteristic of the gap profile, the latter is a field in the advertising payload).

But either way, there can only be one configuration active at a time. You could in theory alternate between two advertising payloads (with different names), but they'd have the same address, so once a central connects to you you wouldn't know which set of services to respond with on discovery.

**Reply by amplatzer** [reply]
thanks for the reply, but according to [this](https://docs.silabs.com/bluetooth/2.13/code-examples/stack-features/adv-and-scanning/multiple-adv-sets-bt5 ) . it says BT 5 supports having more than one advertising sets concurrently.

Was wondering how to implement it on MicroPython .

**Reply by jimmo** [reply, marked as answer]
MicroPython does not (yet) support BLE extended advertising (or any 5.x features).

NimBLE (the host stack used by MicroPython on ESP32) does support some 5.x features (included extended advertising), but none of this is currently used by MicroPython.

FWIW, on ESP32, the 5.x features are only supported at all on C2, C3, S3, H2 (of which only the C3, S3 are supported by MicroPython).

But you asked to be able to have a different set of services for each advertiser -- I don't know if extended advertising lets you do this? Obviously it allows you to _advertise_ different services in the payload, but once connected I don't know what the GATT discovery sees. (I haven't read this part of the spec in detail)
